### PR TITLE
Add NexusStoppingEvent, fire right before NexusStoppedEvent

### PR DESCRIPTION
**Pull request by jdillon**
https://issues.sonatype.org/browse/NEXUS-4795

### CODE REVIEW

File: nexus/nexus-api/src/main/java/org/sonatype/nexus/proxy/events/NexusStoppingEvent.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+
+package org.sonatype.nexus.proxy.events;
+
+/**
+ * Event fired when Nexus has started to shutdown.
+ *
+ * This event is emitted just prior to {@link NexusStoppedEvent} and should be used only
+ * for an early hint that shutdown is going to occur, to stop network connections, etc.
```
Comment (bdemers) [inline nexus/nexus-api/src/main/java/org/sonatype/nexus/proxy/events/NexusStoppingEvent.java:20]: add @since

**Comment by bdemers** [comment]
still might run into potential timing problems between plugins with interdependencies but that would be up to the plugins to deal with I guess

+1

**Comment by jdillon** [comment]
Right, its not the best solution.  A real dependency mechanism between components would be best, but this helps a little and has very little side effects other than being sorta hacky.

**Comment by jdillon** [comment]
BTW, tamas already sorta commented on the change:

  cstamas__: my wee hack looks like this btw: https://gist.github.com/e6c2f6b6375620a20471
  10:50 AM cstamas__
  jdillon: looks good

**Comment by mcculls** [comment]
+1

**Comment by rseddon** [comment]
+1 
Seems simple enough, and since SP will be the only consumer at the moment don't see any reason not to merge it.

**Pull request merged**